python/packages/nisar/pointing/doppler_lut_from_raw.py

```
@@ -417,9 +422,26 @@ def time_dop_est(echo, prf, lag=1, axis=None):
     sr_stop = sr_start + (num_blk_rg - 1) * sr_spacing
     slrg_per_blk = np.linspace(sr_start, sr_stop, num=num_blk_rg)
 
+    # set range line slice of echo
+    rgl_start_stop = [0, tot_pulses]
+    if rangeline_slice is not None:
+        if rangeline_slice.start is not None:
+            rgl_start_stop[0] = min(
+                max(rangeline_slice.start, 0),
+                tot_pulses - 1)
+        if rangeline_slice.stop is not None:
+            rgl_start_stop[1] = max(
+                min(rangeline_slice.stop, tot_pulses),
+                rgl_start_stop[0] + 1)
+    logger.info('[start, stop] range line index to be processed -> '
+                f'{rgl_start_stop}')
+    # update number of range lines to be processed
+    num_rgls = rgl_start_stop[1] - rgl_start_stop[0]
+    logger.info(f'Number of range lines to be processed -> {num_rgls}')
```

Need to throw an exception if `rangeline_slice.step not in (None, 1)`.